packages/@glimmer/reference/lib/template.ts

```diff
@@ -200,6 +200,10 @@ export class HelperRootReference<T = unknown> extends RootReference<T> {
  * recursively calling `get` on the previous reference as a template chain is
  * followed.
  */
+export interface PropertyReference {
```

Comment:
This interface is used to merge the property into the class definition dynamically, without adding a class field. Can you add a comment to that effect, to document this?